### PR TITLE
Add Adkami Presence

### PR DESCRIPTION
**Pull request by inacraft**
- Add Adkami Presence
tzegtzegz
nnjjnjnjn

### CODE REVIEW

File: websites/A/Adkami/dist/presence.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        }
+    }
+    if (presenceData.details == null) {
+        presence.setTrayTitle();
+        presence.setActivity();
+    }
+    else {
+        presence.setActivity(presenceData);
+    }
+});
```
Comment (ririxidev) [inline websites/A/Adkami/dist/presence.js:122]: delete whole this file.

File: websites/A/Adkami/presence.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
+  const presence = new Presence({
+    clientId: "843629449068937217" //The client ID of the Application created at https://discordapp.com/developers/applications
+  });
+
+  var browsingStamp = Math.floor(Date.now() / 1000);
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:5]:
```suggestion
  const presence = new Presence({
    clientId: "843629449068937217"
  }), browsingStamp = Math.floor(Date.now() / 1000);

```

File: websites/A/Adkami/presence.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  var title: any;
+  var button: any;
+
+  presence.on("UpdateData", async () => {
+
+  const presenceData: PresenceData = {
+    largeImageKey: "adkami",
+  }
+    
+  if (document.location.hostname == "www.adkami.com") {
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:15]: don't need this if

File: websites/A/Adkami/presence.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      }
+    }else if(document.location.pathname.includes("/drama") || (document.location.pathname.includes("/video") && document.location.search.includes("t=5"))){
+            if (document.querySelector("#body > section > div.col-12.col-l-9.fiche-look > div.row.blocshadow > div > h1") !== null) {
+        presenceData.startTimestamp = browsingStamp;
+        title = document.querySelector("#body > section > div.col-12.col-l-9.fiche-look > div.row.blocshadow > div > h1")
+        presenceData.details = "Viewing Drama:";
+        presenceData.state = title.innerText;
+             presenceData.buttons = [
+            {
+              label: "Go to the drama",
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:43]:
```suggestion
              label: "View Drama",
```

File: websites/A/Adkami/presence.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      }
+    }else if(document.location.pathname.includes("/hentai") || (document.location.pathname.includes("/video") && document.location.search.includes("t=4"))){
+            if (document.querySelector("#body > section > div.col-12.col-l-9.fiche-look > div.row.blocshadow > div > h1") !== null) {
+        presenceData.startTimestamp = browsingStamp;
+        title = document.querySelector("#body > section > div.col-12.col-l-9.fiche-look > div.row.blocshadow > div > h1")
+        presenceData.details = "Viewing Hentai:";
+        presenceData.state = title.innerText;
+             presenceData.buttons = [
+            {
+              label: "Go to the hentai",
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:75]:
```suggestion
              label: "View Hentai",

```

File: websites/A/Adkami/presence.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    } 
+  }else if(document.location.pathname.includes("/club")){
+     if (document.querySelector("#body > section > div.col-12.col-l-9.blocm.box-display.club-global > h1 > span:nth-child(1)") !== null) {
+        presenceData.startTimestamp = browsingStamp;
+        title = document.querySelector("#body > section > div.col-12.col-l-9.blocm.box-display.club-global > h1 > span:nth-child(1)")
+        presenceData.details = "Viewing club page:";
+        presenceData.state = title.innerText;
+             presenceData.buttons = [
+            {
+              label: "Watch Club Page",
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:91]:
```suggestion
              label: "Check Club Page",

```

File: websites/A/Adkami/presence.ts [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    largeImageKey: "adkami",
+  }
+    
+  if (document.location.hostname == "www.adkami.com") {
+    if (document.location.pathname == "/") {
+      presenceData.startTimestamp = browsingStamp;
+      presenceData.details = "Viewing Home Page";
+    }else if(document.location.pathname.includes("/anime") || (document.location.pathname.includes("/video") && document.location.search.includes("t=0"))){
+      if (document.querySelector("#body > section > div.col-12.col-l-9.fiche-look > div.row.blocshadow > div > h1") !== null) {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:21]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Viewing Anime:";
+        presenceData.state = title.innerText;
+             presenceData.buttons = [
+            {
+              label: "Go to the anime",
+              url: document.URL
+            }
+          ];
+      }else{
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:32]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Looking for animes";
+      }
+    }else if(document.location.pathname.includes("/drama") || (document.location.pathname.includes("/video") && document.location.search.includes("t=5"))){
+            if (document.querySelector("#body > section > div.col-12.col-l-9.fiche-look > div.row.blocshadow > div > h1") !== null) {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:37]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Viewing Drama:";
+        presenceData.state = title.innerText;
+             presenceData.buttons = [
+            {
+              label: "Go to the drama",
+              url: document.URL
+            }
+          ];
+      }else{
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:48]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Viewing Serie:";
+        presenceData.state = title.innerText;
+             presenceData.buttons = [
+            {
+              label: "Go to the serie",
+              url: document.URL
+            }
+          ];
+      }else{
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:64]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Looking for series";
+      }
+    }else if(document.location.pathname.includes("/hentai") || (document.location.pathname.includes("/video") && document.location.search.includes("t=4"))){
+            if (document.querySelector("#body > section > div.col-12.col-l-9.fiche-look > div.row.blocshadow > div > h1") !== null) {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:69]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+              url: document.URL
+            }
+          ];
+      }else{
+        presenceData.startTimestamp = browsingStamp;
+        presenceData.details = "Looking for dramas";
+      }
+    }else if(document.location.pathname.includes("/serie") || (document.location.pathname.includes("/video") && document.location.search.includes("t=1"))){
+            if (document.querySelector("#body > section > div.col-12.col-l-9.fiche-look > div.row.blocshadow > div > h1") !== null) {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:53]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Viewing Hentai:";
+        presenceData.state = title.innerText;
+             presenceData.buttons = [
+            {
+              label: "Go to the hentai",
+              url: document.URL
+            }
+          ];
+      }else{
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:80]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Looking for juicy contents";
+    } 
+  }else if(document.location.pathname.includes("/club")){
+     if (document.querySelector("#body > section > div.col-12.col-l-9.blocm.box-display.club-global > h1 > span:nth-child(1)") !== null) {
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:85]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Viewing club page:";
+        presenceData.state = title.innerText;
+             presenceData.buttons = [
+            {
+              label: "Watch Club Page",
+              url: document.URL
+            }
+          ];
+     }else{
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:96]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Browsing Club";
+     }
+  }else if(document.location.pathname.includes("/news")){
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:100]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Browsing News";
+  }else{
+    presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:103]:
```suggestion
```

File: websites/A/Adkami/presence.ts [hunk 19]
```diff
@@ -0,0 +1,114 @@
...
+
+  var browsingStamp = Math.floor(Date.now() / 1000);
+  var title: any;
+  var button: any;
+
+  presence.on("UpdateData", async () => {
+
+  const presenceData: PresenceData = {
+    largeImageKey: "adkami",
+  }
```
Comment (ririxidev) [inline websites/A/Adkami/presence.ts:13]:
```suggestion
  const presenceData: PresenceData = {
    largeImageKey: "adkami",
    startTimestamp: browsingStamp
  }
```

**Review by ririxidev — CHANGES_REQUESTED** [review]
and maybe add support for player?

**Comment by inacraft** [comment]
> and maybe add support for player?

I did all the requested changes and for player, i actually want to do that but when i see that the site have more than 1 video player, these can change and i think it's use iframe, it's a little bit hard. Because, it's my first time doing a presence I can continue to try to make it one but I think it's going to be a little difficult ^^"

**Comment by inacraft** [comment]
And thanks for checking/updating my code and correct my English because I am not very good at it :)

File: websites/A/Adkami/dist/metadata.json [hunk 20]
```diff
@@ -0,0 +1,22 @@
+{
+  "service": "Adkami",
+  "author": {
+    "name": "Inacraft",
+    "id": "1929"
+  },
+  "category": "anime",
+  "logo": "https://www.adkami.com/favicon2.png",
```
Comment (Bas950) [inline websites/A/Adkami/dist/metadata.json:8]: upload to imgur

File: websites/A/Adkami/dist/metadata.json [hunk 21]
```diff
@@ -0,0 +1,22 @@
+{
+  "service": "Adkami",
+  "author": {
+    "name": "Inacraft",
+    "id": "1929"
```
Comment (Bas950) [inline websites/A/Adkami/dist/metadata.json:5]: Thats not your Discord id...

File: websites/A/Adkami/dist/metadata.json [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+  "color": "#7ccd18",
+  "tags": [
+    "animes",
+    "nsfw"
+  ],
+  "url": "www.adkami.com",
+  "description": {
+    "fr": "ADKami est un site qui a pour objectif d'aider les gens a suivre les sortie des episodes d'animes sur les plateforme legal.",
+    "en": "ADKami is a site that aims to help people follow the release of anime episodes on the legal platform."
+  },
```
Comment (Bas950) [inline websites/A/Adkami/dist/metadata.json:20]:
```suggestion
  "description": {
    "fr": "ADKami est un site qui a pour objectif d'aider les gens a suivre les sortie des episodes d'animes sur les plateforme legal.",
    "en": "ADKami is a site that aims to help people follow the release of anime episodes on the legal platform.",
    "nl": "ADKami is een site die mensen wil helpen de releases van anime-afleveringen op een legale platform te volgen."
  },

```

File: websites/A/Adkami/presence.ts [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+        presenceData.state = title.innerText;
+             presenceData.buttons = [
+            {
+              label: "View Hentai",
+
+              url: document.URL
+            }
+          ];
+      }else{
+        presenceData.details = "Looking for juicy contents";
```
Comment (Bas950) [inline websites/A/Adkami/presence.ts:70]: please add some more appropriate text..

File: websites/A/Adkami/presence.ts [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
+              url: document.URL
+            }
+          ];
+     }else{
+        presenceData.details = "Browsing Club";
+     }
+  }else if(document.location.pathname.includes("/news")){
+        presenceData.details = "Browsing News";
+  }else{
+    presenceData.details = "Surfing on the site";
```
Comment (Bas950) [inline websites/A/Adkami/presence.ts:90]:
```suggestion
    presenceData.details = "Browsing...";
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/Adkami/dist/metadata.json (hunk 21 above)
Comment (inacraft) [reply websites/A/Adkami/dist/metadata.json]: My Bad, I confused

File: websites/A/Adkami/presence.ts (hunk 23 above)
Comment (inacraft) [reply websites/A/Adkami/presence.ts]: I replaced all nsfw things by Anime and adding (-18) and it's more appropriate.

**Comment by ririxidev** [comment]
for now, presence doesn't offer a lot to be as presence in our store sorry

**Pull request closed without merging**